src/Converter/Converter.php

```diff
@@ -244,9 +250,13 @@ protected function getConverterContext(): ConverterContext
     /**
      * Run all plugins to convert a value.
      */
-    protected function fromSqlUsingPlugins(null|int|float|string|object $value, string $type, ?string $realType = null): mixed
+    protected function fromSqlUsingPlugins(null|int|float|string|object $value, ?string $type, ?string $realType = null): mixed
     {
         $realType ??= $type;
+        if (!$realType) {
+            throw new QueryBuilderError();
```

Comment:
No message?

Reply:
Really there's no need here, this is internal code that isn't supposed to be called from outside: the error case would simply mean the calling code is wrong.